src/Auditable.php

```diff
@@ -282,18 +283,24 @@ public function toAudit(): array
 
         $tags = implode(',', $this->generateTags());
 
-        return $this->transformAudit([
+        $data = [
             'old_values'     => $old,
             'new_values'     => $new,
             'event'          => $this->auditEvent,
             'auditable_id'   => $this->getKey(),
             'auditable_type' => $this->getMorphClass(),
-            $userForeignKey  => $this->resolveUser(),
+            $userForeignKey  => $this->resolveUserId(),
             'url'            => $this->resolveUrl(),
             'ip_address'     => $this->resolveIpAddress(),
             'user_agent'     => $this->resolveUserAgent(),
             'tags'           => empty($tags) ? null : $tags,
-        ]);
+        ];
+
+        if (Config::get('audit.morphable', true)) {
+            $data['user_type'] = $this->resolveUserClass();
```

Comment:
And replace `$data['user_type'] = $this->resolveUserClass();` with

`$data['user_type'] = $user->getMorphClass();`